src/adams_utils.jl

```diff
@@ -1,42 +1,49 @@
-function ϕ_and_ϕstar!(cache, dy, next_point, last_idx)
-  @unpack grid_points, ϕstar_nm1, ϕ_n, ϕstar_n,β,k = cache
-  if k < 3
-    if typeof(dy) <: Array
-      ϕ_n = Array{Array{eltype(dy)}}(1,k)
-      ϕstar_n = Array{Array{eltype(dy)}}(1,k)
+function ϕ_and_ϕstar!(cache, du)
```

Comment:
can you put a comment for the reference of this function and its implementation?